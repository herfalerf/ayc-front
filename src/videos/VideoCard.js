import React from "react";

// Display embedded youtube video
//
// Is rendered by VideoList to a show a "card" for each video.
//
// VideoList -> VideoCard

function VideoCard({ link, name, username }) {
  console.debug("VideoCard", link);

  return (
    <div>
      <iframe
        width="560"
        height="315"
        src={link}
        title="YouTube video player"
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
      ></iframe>
      {name && <h3> {name}</h3>}
    </div>
  );
}

export default VideoCard;
