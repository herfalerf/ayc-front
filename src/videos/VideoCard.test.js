import React from "react";
import { render } from "@testing-library/react";
import VideoCard from "./VideoCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(<VideoCard />);
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <VideoCard link="video.youtube.com" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
