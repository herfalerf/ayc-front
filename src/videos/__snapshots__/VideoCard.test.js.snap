// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-playerWrapper-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="makeStyles-reactPlayer-4"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
